iothub_client/inc/iothub_device_client.h

```diff
@@ -296,6 +297,46 @@ extern "C"
     */
     MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubDeviceClient_GetTwinAsync, IOTHUB_DEVICE_CLIENT_HANDLE, iotHubClientHandle, IOTHUB_CLIENT_DEVICE_TWIN_CALLBACK, deviceTwinCallback, void*, userContextCallback);
 
+    /**
+    * @brief    This API provides a way to retrieve the device twin desired properties on-demand.
+    *
+    * @param    iotHubClientHandle      The handle created by a call to the hub client create function.
+    * @param    twinRequestOptions      Pointer to the twin request options struct. Can only be
+    *                                   used when requesting desired or reported sections.
+    * @param    deviceTwinDesiredCallback   The callback specified by the device client to receive
+    *                                       the  twin document. If any failures occur, the callback
+    *                                       is invoked passing @c NULL as twinResponse and as
+    *                                       payload, and zero as size.
+    * @param    userContextCallback     User specified context that will be provided to the
+    *                                   callback. This can be @c NULL.
+    *
+    *            @b NOTE: The application behavior is undefined if the user calls
+    *            the IoTHubDeviceClient_Destroy function from within any callback.
```

Comment:
doxygen will link to the function's doc if you put `()` at the end of the function name
```suggestion
    *            the IoTHubDeviceClient_Destroy() function from within any callback.
```